core-container/entry.sh

```diff
@@ -242,51 +230,68 @@ function run_kcp() {
 
 function run_init() {
     echo "--< Starting init >--"
-    wait_kcp_ready
     set_provider_adapters
-    kubestellar init --local-kcp false --ensure-imw $ENSURE_IMW --ensure-wmw $ENSURE_WMW
-    kubectl ws root
+    kubestellar init --ensure-imw $ENSURE_IMW --ensure-wmw $ENSURE_WMW 
     touch ready
     echo "***READY***"
     sleep infinity
 }
 
-
 function run_mailbox_controller() {
     echo "--< Starting mailbox-controller >--"
     wait-kubestellar-ready
+    # TODO: remove the kcp dependency
+    get_kcp_kubeconfig
+    KUBECONFIG=$kcp_kubeconfig
     kubectl ws root:espw
     if ! mailbox-controller -v=${VERBOSITY} ; then
         echoerr "unable to start mailbox-controller!"
         exit 1
     fi
 }
 
-
 function run_where_resolver() {
     echo "--< Starting where-resolver >--"
     wait-kubestellar-ready
+    # TODO: remove the kcp dependency
+    get_kcp_kubeconfig
+    KUBECONFIG=$kcp_kubeconfig
     kubectl ws root:espw
     if ! kubestellar-where-resolver -v ${VERBOSITY} ; then
         echoerr "unable to start kubestellar-where-resolver!"
         exit 1
     fi
 }
 
-
 function run_placement_translator() {
     echo "--< Starting placement-translator >--"
     wait-kubestellar-ready
+    # TODO: remove the kcp dependency
+    get_kcp_kubeconfig
+    KUBECONFIG=$kcp_kubeconfig
     kubectl ws root:espw
     if ! placement-translator --allclusters-context  "system:admin" -v=${VERBOSITY} ; then
         echoerr "unable to start mailbox-controller!"
         exit 1
     fi
 }
 
+# get_host_kubeconfig: 
```

Comment:
"get" is the wrong word to use in the name of a function that modifies something.